uamqp/management_link.py

```diff
@@ -4,92 +4,185 @@
 # license information.
 #--------------------------------------------------------------------------
 
-import threading
-import struct
-import uuid
 import logging
-import time
-from urllib.parse import urlparse
-from enum import Enum
-from io import BytesIO
+from collections import namedtuple
 
-from .endpoints import Source, Target
+from .sender import SenderLink
+from .receiver import ReceiverLink
 from .constants import (
-    DEFAULT_LINK_CREDIT,
-    SessionState,
-    SessionTransferState,
     ManagementLinkState,
-    LinkDeliverySettleReason,
     LinkState,
-    Role,
     SenderSettleMode,
-    ReceiverSettleMode
-)
-from .performatives import (
-    AttachFrame,
-    DetachFrame,
-    TransferFrame,
-    DispositionFrame,
-    FlowFrame,
+    ReceiverSettleMode,
+    ManagementExecuteOperationResult,
+    ManagementOpenResult,
+    SEND_DISPOSITION_REJECT
 )
 
 _LOGGER = logging.getLogger(__name__)
 
+PendingMgmtOperation = namedtuple('PendingMgmtOperation', ['message', 'on_execute_operation_complete'])
+
 
 class ManagementLink(object):
     """
 
     """
-    def __init__(self, session, endpoint, **kwargs):
+    def __init__(
+            self,
+            session,
+            endpoint,
+            status_code_field=b'statusCode',
+            status_description_field=b'statusDescription',
```

Comment:
These two should be moved into kwargs.
They only exist to account for a bug in the AMQP spec/service implementation